sklearn/feature_selection/_rfe.py

```diff
@@ -286,6 +287,14 @@ def _fit(self, X, y, step_score=None, **fit_params):
             n_features_to_select = n_features // 2
         elif isinstance(self.n_features_to_select, Integral):  # int
             n_features_to_select = self.n_features_to_select
+            if n_features_to_select > n_features:
+                warnings.warn(
+                    (
+                        "n_features_to_select must not exceed available features."
+                        " Falling back to select all n_features."
```

Comment:
the "must not" would be appropriate for an error but feels weird when it's just a warning. What do you think of this rephrasing ?
```suggestion
                        f"Found {n_features_to_select=} > {n_features=}."
                        " There will be no feature selection and all features will be kept."
```